### PR TITLE
Fix irsa for k8s < 1.20

### PR DESCRIPTION
**Pull request by olemarkus**
IRSA on 1.19 segfaults because we are not setting the feature gate that the aws model expects.

### CODE REVIEW

**Comment by hakman** [comment]
@olemarkus Could you split this into 2 commits so that at least the main one would be easy to track in various branches?

File: pkg/model/components/apiserver.go [hunk 1]
```diff
@@ -198,6 +198,12 @@ func (b *KubeAPIServerOptionsBuilder) BuildOptions(o interface{}) error {
 		}
 	}
 
+	if b.IsKubernetesLT("1.20") && clusterSpec.ServiceAccountIssuerDiscovery != nil && fi.BoolValue(&clusterSpec.ServiceAccountIssuerDiscovery.EnableAWSOIDCProvider) {
```
Comment (hakman) [inline pkg/model/components/apiserver.go:201]: Do we need a minimum version here too, in case we want to cherry-pick this to other release branches?

Comment (olemarkus) [reply pkg/model/components/apiserver.go:201]: I added some validation to cover this case.

File: pkg/apis/kops/validation/legacy.go [hunk 2]
```diff
@@ -375,28 +375,39 @@ func ValidateCluster(c *kops.Cluster, strict bool) field.ErrorList {
...
+					}
+				default:
 					allErrs = append(allErrs, field.Invalid(saidStoreField, saidStore, "S3 is the only supported VFS for discoveryStore"))
 				}
-			default:
-				allErrs = append(allErrs, field.Invalid(saidStoreField, saidStore, "S3 is the only supported VFS for discoveryStore"))
 			}
 		}
-	}
+		if said.EnableAWSOIDCProvider {
```
Comment (hakman) [inline pkg/apis/kops/validation/legacy.go:400]: Let's just add this a as separate block, if you don't mind.
```suggestion
		if said != nil && said.EnableAWSOIDCProvider {
```

Comment (olemarkus) [reply pkg/apis/kops/validation/legacy.go]: The said stuff was getting too big. Moved to a dedicated function.

**Review by hakman — APPROVED** [review]

**Pull request merged**